clang/test/CIR/Lowering/union.cir

```diff
@@ -0,0 +1,10 @@
+// RUN: cir-opt %s --cir-to-llvm | FileCheck %s
+
+!u_empty = !cir.record<union {}>
```

Comment:
Can you add a check for the LLVM IR type here? I think we're probably generating the wrong thing. In the incubator, we generate this:

`%union.U = type {}`

Whereas classic codegen generates this:

`%union.U = type { [4 x i8] }`

Reply:
Let me amend what I said above. I didn't realize the classic codegen I was referring to was using the Windows C ABI. Classic codegen does generate `%union.U = type {}` for C and `%union.U = type { i8 }` for C++ (which the incubator also does).

So, I guess my question is whether we identify these padded forms as "empty".